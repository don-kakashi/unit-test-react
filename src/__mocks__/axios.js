const axiosMock = jest.genMockFromModule('axios');

const mockResponse = {
  data: {
    shops: [{
      location: 'test location',
      address: 'test address',
    }]
  }
};

axiosMock.get.mockImplementation(req);

function req() {
  return new Promise(resolve =>
    axiosMock.delayTimer = setTimeout(() => {
        resolve(mockResponse)
      },
      100
    )
  )
}

module.exports = axiosMock;