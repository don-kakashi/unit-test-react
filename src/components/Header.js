
import React from 'react';

export default () => 
  <img
    src="images/wired-brain-coffee-logo.png"
    alt="logo"
  />