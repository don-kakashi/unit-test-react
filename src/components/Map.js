import React from 'react';

export default ({imageName}) =>
  <img src={`images/${imageName || 'none.png'}`} alt={imageName || `no-map`}/>