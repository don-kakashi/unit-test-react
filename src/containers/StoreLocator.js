import React, { Component } from 'react';
import Header from '../components/Header';
import Button from '../components/Button';
import Map from '../components/Map';
import mapChooser from '../mapChooser';
import Axios from 'axios';

export default class extends Component{
  constructor(props) {
    super(props)
    this.state = {
      shops: [],
      currentMap: 'none.png'
    }
  }

  async componentDidMount() {
    const { data: { shops }} = await Axios.get('/data/shops.json');
    this.setState({ shops });
  }

  chooseMap = e => {
    this.setState({
      currentMap: mapChooser(e.target.value)
    })
  }

  render () {
    const { location } = this.props;
    const { shops, currentMap } = this.state;
    return <>
      <Header />
      <div>
        { shops.map(({location}, idx) =>
          <Button key={idx} location={location} handleClick={this.chooseMap} /> 
        )}
      </div>
      <Map imageName={currentMap} location={location}/>
    </>
  }
}