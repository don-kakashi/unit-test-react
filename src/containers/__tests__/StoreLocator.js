import React from 'react'
import { shallow } from 'enzyme';
import axios from 'axios';
import StoreLocator from '../StoreLocator';
import Header from '../../components/Header';
import Map from '../../components/Map';
import renderer from 'react-test-renderer'

describe('StoreLocator', () => {
  let mountedStoreLocator;
  beforeEach(() => {
    mountedStoreLocator = shallow( <StoreLocator /> );
  });

  it('Should call axios.get with the correct url', () => {
    return mountedStoreLocator.instance().componentDidMount(()=>
      expect(axios.get).toHaveBeenCalledWith('/data/shops.json')
    )
  })

  it('Should update state with api data', () => {
    return mountedStoreLocator.instance().componentDidMount(()=>
      expect(mountedStoreLocator.state()).toHaveProperty('shops', [{
        location: 'test location',
        address: 'test address',
      }]))
  })

  it('Should render without crashing', () => {
    const tree = renderer.create(<StoreLocator />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should render a Header', () => {
    const locators = mountedStoreLocator.find(Header);
    expect(locators.length).toBe(1);
  });

  it('Should render a Map', () => {
    const locators = mountedStoreLocator.find(Map);
    expect(locators.length).toBe(1);
  });
});

describe('chooseMap', () => {
  it('Should update the state "currentMap" using the location passed to it', () => {
    const mountedStoreLocator = shallow( <StoreLocator /> );
    const mockEvent = {target:{value:'testland'}};
    mountedStoreLocator.instance().chooseMap(mockEvent);
    expect(mountedStoreLocator.instance().state.currentMap).toBe('testland.png')
  });
})