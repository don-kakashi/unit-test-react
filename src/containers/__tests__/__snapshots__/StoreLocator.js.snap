// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoreLocator Should render without crashing 1`] = `
Array [
  <img
    alt="logo"
    src="images/wired-brain-coffee-logo.png"
  />,
  <div />,
  <img
    alt="none.png"
    src="images/none.png"
  />,
]
`;
